src/openms/source/ANALYSIS/OPENSWATH/OpenSwathOSWWriter.cpp

```diff
@@ -153,7 +153,8 @@ namespace OpenMS
     // Insert run_id information
     std::stringstream sql_run;
     sql_run << "INSERT INTO RUN (ID, FILENAME) VALUES ("
-            << *(int64_t*)&run_id_ << ", '" // Conversion from UInt64 to int64_t to support SQLite
+            // Conversion from UInt64 to int64_t to support SQLite (and conversion to 63 bits)
+            <<  static_cast<int64_t >(run_id_ & ~(1UL << 63)) << ", '"
```

Comment:
Why do you need this?
It run_id_ potentially larger than 2^63? Otherwise a simple cast would do, right?

Reply:
the issue is that the OpenMS UniqueId is 64 bits unsigned while SQLite interprets all values as signed values, so when you retrieve the values in Python they are negative in half the cases. Its ugly but it means the values are not comparable between C++ and Python, this should solve the problem.

Reply:
ok, I see.
If you do not care about the actual value, then right-shifting by 1 would do the same thing. Might be easier to reason about.